[docs] update endpoint documentation (#4060)
sveltejs/kit: documentation/docs/01-routing.md

> All server-side code, including endpoints, has access to `fetch` in case you need to request data from external APIs. Don't worry about the `$lib` import, we'll get to that [later](/docs/modules#$lib).

> This function takes a [`RequestEvent`](types#sveltejs-kit-requestevent) and typically returns an [`EndpointOutput`](types#sveltejs-kit-endpointoutput) (`{ status, headers, body }`) object representing the response, where `status` is an [HTTP status code](https://httpstatusdogs.com):

Reviewer: this doesn't render nicely:

<img width="239" alt="image" src="https://user-images.githubusercontent.com/1162160/155218982-002ce539-4b57-4160-8746-97c37be25e50.png">

and 'typically' is confusing — in what circumstances is this true or not true? I think we should just link to the type definition

```suggestion
The job of a [request handler](/docs/types#sveltejs-kit-requesthandler) is to return a `{ status, headers, body }` object representing the response, where `status` is an [HTTP status code](https://httpstatusdogs.com):
```

> This function takes a [`RequestEvent`](types#sveltejs-kit-requestevent) and typically returns an [`EndpointOutput`](types#sveltejs-kit-endpointoutput) (`{ status, headers, body }`) object representing the response, where `status` is an [HTTP status code](https://httpstatusdogs.com):

Author: the reason I linked to the input and output is that it saves a click whereas you can't tell much from the `RequestHandler` docs alone since you need to go to these other docs to understand the types. not a big deal though and I'm happy to just leave it as `RequestHandler`

"typically" is indeed a bit imprecise. Though I think the current / suggested docs may not be completely accurate. If I'm reading the docs correctly, `RequestHandler` can return either an `EndpointOutput` or a `Response`

> This function takes a [`RequestEvent`](types#sveltejs-kit-requestevent) and typically returns an [`EndpointOutput`](types#sveltejs-kit-endpointoutput) (`{ status, headers, body }`) object representing the response, where `status` is an [HTTP status code](https://httpstatusdogs.com):

Reviewer: a standalone endpoint can return a `Response` by virtue of that being a valid `EndpointOutput` — i.e. it has a `status`, `headers` and `body`
